--- check_ndrestart    2020-09-16 17:18:13.874362859 +0000
+++ check_ndrestart-AWS  2020-09-16 17:19:06.254972929 +0000
@@ -60,11 +60,6 @@
         exit $STATE_UNKNOWN
 fi

-if [ ! $(rpm -q redhat-lsb-core) ]; then
-        echo "UNKNOWN: the required package redhat-lsb-core is not installed on this system".
-        exit $STATE_UNKNOWN
-fi
-
 if [ ! -x $LSBRELEASE ]; then
         echo "UNKNOWN: lsb_release not found or is not executable by the nagios/icinga user."
         exit $STATE_UNKNOWN
@@ -133,6 +128,16 @@


 case $LSBR_DISTRID in
+    Amazon)
+        NEEDREBOOT=$(needs-restarting -r 2>&1)
+        RC_NEEDREBOOT=$?
+
+        if [ -n "${EXCLUDE}" ]; then
+            NEEDSRVRESTART=$(sudo needs-restarting -s 2>&1 | egrep -v "${EXCLUDE}")
+                    else
+            NEEDSRVRESTART=$(sudo needs-restarting -s 2>&1)
+        fi
+        ;;
     RedHatEnterpriseServer | CentOS)
         if [ $(bc <<< "$LSBR_DISTRRN >= 7.3") -ne 0 ]; then
             NEEDREBOOT=$(needs-restarting -r 2>&1)
